internal/scripts/scripts.go

```diff
@@ -33,6 +37,7 @@ type ContentValue struct {
 	RootDir    string
 	CheckRead  func(path string) error
 	CheckWrite func(path string) error
+	CreateFile func(opts *fsutil.CreateOptions) error
```

Comment:
What's the long term view here? What about everything else?

Reply:
We are replacing `os.WriteFile` with out `fsutil.Create`, the problem is that the latter can create directories which is outside the capabilities of scripts as of now. My idea was to match the existing implementation and error check that in fact we are not creating directories.

In the future when we can create them, we can change the function to `Create` and use it for both. My reasoning was to have it as tight as possible right now in order to avoid unexpected bugs.

Reply:
Discussed that during the meeting. We have changed the API to only be concerned about reporting mutated files. This is because:
1. We do not have to create a callback for each operation we perform on a file (e.g. appending to a file without reading it).
2. We can keep report local to slicer and the scripts package does not need to know about it.
3. It has the greatest flexibility because we can add low level primitives for file manipulation in scripts without changes to the interface or ContentValue becoming a kitchen sink.
4. It is more coherent because previously we were only "overriding" the file creation part, not the file reading or the directory listing. And the reason for it was not clear in the code which leads to poor readability.

At the same time, we don't have more concrete use cases or optimizations now so we could not even design the comprehensive API if we wanted to, and it does not make sense to invest in a design that is going to change quickly.